systems/controllers/cp_traj_gen.cc

```diff
@@ -10,77 +10,76 @@ using std::cout;
 using std::endl;
 using std::string;
 
+using Eigen::MatrixXd;
 using Eigen::Vector2d;
 using Eigen::Vector3d;
 using Eigen::VectorXd;
-using Eigen::MatrixXd;
 
+using drake::systems::BasicVector;
 using drake::systems::Context;
-using drake::systems::DiscreteValues;
 using drake::systems::DiscreteUpdateEvent;
+using drake::systems::DiscreteValues;
 using drake::systems::EventStatus;
-using drake::systems::BasicVector;
 
-using drake::trajectories::PiecewisePolynomial;
 using drake::trajectories::ExponentialPlusPiecewisePolynomial;
+using drake::trajectories::PiecewisePolynomial;
 
 namespace dairlib {
 namespace systems {
 
-CPTrajGenerator::CPTrajGenerator(const RigidBodyTree<double>& tree,
-                                 double mid_foot_height,
-                                 double desired_final_foot_height,
-                                 double desired_final_vertical_foot_velocity,
-                                 double max_CoM_to_CP_dist,
-                                 double stance_duration_per_leg,
-                                 int left_foot_idx,
-                                 Eigen::Vector3d pt_on_left_foot,
-                                 int right_foot_idx,
-                                 Eigen::Vector3d pt_on_right_foot,
-                                 int pelvis_idx,
-                                 bool add_extra_control,
-                                 bool is_feet_collision_avoid,
-                                 bool is_using_predicted_com,
-                                 double cp_offset,
-                                 double center_line_offset) :
-    tree_(tree),
-    mid_foot_height_(mid_foot_height),
-    desired_final_foot_height_(desired_final_foot_height),
-    desired_final_vertical_foot_velocity_(desired_final_vertical_foot_velocity),
-    max_CoM_to_CP_dist_(max_CoM_to_CP_dist),
-    stance_duration_per_leg_(stance_duration_per_leg),
-    left_foot_idx_(left_foot_idx),
-    right_foot_idx_(right_foot_idx),
-    pt_on_left_foot_(pt_on_left_foot),
-    pt_on_right_foot_(pt_on_right_foot),
-    pelvis_idx_(pelvis_idx),
-    add_extra_control_(add_extra_control),
-    is_feet_collision_avoid_(is_feet_collision_avoid),
-    is_using_predicted_com_(is_using_predicted_com),
-    cp_offset_(cp_offset),
-    center_line_offset_(center_line_offset) {
+CPTrajGenerator::CPTrajGenerator(
```

Comment:
This constructor is getting really big, is there a way to condense down the number of parameters? To start, we could probably remove some of the bools (is_feet_collision_avoid). Some parameters like mid_foot_height, max_CoM_to_CP_dist could be set as const static member variables instead.

Reply:
I think all of the parameters are essential and shouldn't be hard-coded, but I combine the right and left leg index/position and states of the finite state machine.

Reply:
After face-to-face discussion with @yangwill, we decided to leave it as is